categories-images.php

```diff
@@ -198,12 +198,34 @@ function zSaveTaxonomyImage($term_id) {
         }
     }
 
-    // get attachment ID by image url
+    /**
+     * Get attachment ID by image url
+     *
+     * @param string $image_src
+     * @return string|null
+     */
     function zGetAttachmentIdByUrl($image_src) {
+
         global $wpdb;
-        $query = $wpdb->prepare("SELECT ID FROM $wpdb->posts WHERE guid = %s", $image_src);
-        $id = $wpdb->get_var($query);
-        return (!empty($id)) ? $id : NULL;
+
+        // Convert and sanitize image url into md5 string.
+        $md5_image_src = md5( esc_url_raw( $image_src ) );
```

Comment:
Fatto!

Reply:
@HyprCoder I think that is not a problem because the maximum key size for Redis is 512MB:

https://redis.io/topics/data-types-intro#redis-keys

What do you think?